src/features/errors/pages/AsyncErrorBoundary.tsx

```diff
@@ -14,21 +14,43 @@ import { ArrowPrevious } from 'ui/svg/icons/ArrowPrevious'
 import { ColorsEnum, getSpacing, Spacer, Typo } from 'ui/theme'
 import { useCustomSafeInsets } from 'ui/theme/useCustomSafeInsets'
 
-export const RetryBoundaryWithNavigation = ({ resetErrorBoundary }: FallbackProps) => {
+export class AsyncError extends Error {
+  public retry?: () => Promise<unknown>
+  constructor(message: string, retry?: () => Promise<unknown>) {
+    super(message)
+    this.retry = retry
+  }
+}
+
+interface AsyncFallbackProps extends FallbackProps {
+  resetErrorBoundary: (...args: Array<unknown>) => void
+  error: AsyncError
+  backNavigation?: boolean
+}
+
+export const AsyncErrorBoundary = ({
+  resetErrorBoundary,
+  error,
+  backNavigation = true,
+}: AsyncFallbackProps) => {
   const { reset } = useQueryErrorResetBoundary()
   const { canGoBack, goBack } = useNavigation<UseNavigationType>()
   const { top } = useCustomSafeInsets()
 
-  const handleRetry = () => {
+  const handleRetry = async () => {
     reset()
     resetErrorBoundary()
+    if (error?.retry) {
+      await error.retry()
+    }
   }
 
   return (
     <Container>
       <Background />
+      <Spacer.TopScreen />
       <Spacer.Flex />
-      {canGoBack() && (
+      {backNavigation && canGoBack() && (
```

Comment:
J'ai l'impression qu'il y a de la redondance ici 
```suggestion
      {canGoBack() && (
```

Reply:
`canGoBack()` est une methode qui vérifie le stack de l'historique et return `true` si `history.length > 1`, alors que `backNavigation` est une `props` pour désactiver le bouton retour de façon stricte, ceci dans le cadre du refactor des ErrorBoundary.